Remove special case for stdlib in IsNonUserModuleRequest

hamishknight opened the pull request:
It's possible this may be the main module, or a "local" dependency within the same project, and should be treated as a user module in such cases.

hamishknight commented:
@swift-ci please test

hamishknight commented:
@swift-ci please build toolchain

bnbarham commented on test/SourceKit/CursorInfo/Inputs/Swift.swift:
```diff
@@ -0,0 +1,4 @@
+// This is built with '-parse-stdlib', so we can't use any stdlib types.
```
Could you use `split-file` instead 🙏 I find it makes these so much easier to follow outside of a review (ie. when trying to work out what a test does after the fact).

hamishknight commented:
@swift-ci please smoke test

hamishknight committed:
Remove special case for stdlib in IsNonUserModuleRequest

It's possible this may be the main module, or a
"local" dependency within the same project, and
should be treated as a user module in such cases.

hamishknight commented:
@swift-ci please smoke test
